miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/ui/MiniAppCustomPermissionWindow.kt

```diff
@@ -0,0 +1,148 @@
+package com.rakuten.tech.mobile.miniapp.permission.ui
+
+import android.app.Activity
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.TextView
+import android.widget.Toast
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.rakuten.tech.mobile.miniapp.R
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermission
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionCache
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionResult
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * A class to show default custom permissions UI to manage permissions in this SDK.
+ */
+internal class MiniAppCustomPermissionWindow(
+    private val activity: Activity,
+    private val customPermissionCache: MiniAppCustomPermissionCache
+) : CoroutineScope {
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Main
+
+    @VisibleForTesting
+    lateinit var customPermissionAlertDialog: AlertDialog
+
+    private lateinit var customPermissionAdapter: MiniAppCustomPermissionAdapter
+    private lateinit var customPermissionLayout: View
+
+    fun displayPermissions(
+        miniAppId: String,
+        permissionsWithDescription: List<Pair<MiniAppCustomPermissionType, String>>
+    ) {
+        if (miniAppId.isEmpty())
+            return
+
+        val deniedPermissions = getDeniedPermissions(miniAppId, permissionsWithDescription)
```

Comment:
I just noticed this implementation doesn't match iOS. We should filter the permissions before passing them to be displayed. So the Host App should only receive a list of denied permissions, not the ones which are already accepted.

If all of the permissions are already accepted, we shouldn't even call the Host App's implementation of `requestCustomPermissions`. In that case, we should just automatically return success to the mini app.

So basically, I think you should move this `getDeniedPermissions` functionality into [MiniAppMessageBridge.onRequestCustomPermissions](https://github.com/rakutentech/android-miniapp/blob/c63ee5683a1300a0fe498a7cbcb381230a4cf03c/miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt#L167), and also prevent the Host App's `requestCustomPermissions` function from being called in the case that all of the permissions are already allowed. See the [iOS implementation](https://github.com/rakutentech/ios-miniapp/blob/d0719004c4b4057a5f673d4a94ec2db786224b82/MiniApp/Classes/JavascriptBridge/MiniAppJavaScriptPermissionManager.swift#L19) for details.

Reply:
You are right! 
That's something we need to inside SDK. I'll update that issue in this PR.